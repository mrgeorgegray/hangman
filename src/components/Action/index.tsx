/** @jsxImportSource @emotion/react */
import React from "react";
import { css } from "@emotion/react";

interface ActionProps {
  onClick: () => void;
  text: string;
  type: "error" | "success" | "warning";
}

const Action: React.FC<ActionProps> = ({ onClick, text, type }) => (
  <button
    css={css`
        background-color: #f5f5f5;
        border: 1px solid #747474;
        border-radius: 3px;
        font-size: 20px;
        padding: 12px 18px;
        margin-right: 20px;
        transition: all 150ms ease-in;

        &:hover {
          color: white;
          cursor: pointer;
          ${
            type === "error" &&
            `
              background-color: #ff0033;
              border-color: #ff0033;
            `
          }
          ${
            type === "warning" &&
            `
              background-color: #ff6633;
              border-color: #ff6633;
            `
          }
          ${
            type === "success" &&
            `
              background-color: #6cc070;
              border-color: #6cc070;
            `
          }
        },
      `}
    onClick={onClick}
  >
    {text}
  </button>
);

export default Action;
