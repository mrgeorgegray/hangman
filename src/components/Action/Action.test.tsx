import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Action from ".";

describe("<Action />", () => {
  const onClick = jest.fn();
  const text = "Click me";

  it("renders", () => {
    const { container } = render(<Action onClick={onClick} text={text} />);
    expect(container).toMatchSnapshot();
  });

  it("handles clicks", () => {
    const { getByText } = render(<Action onClick={onClick} text={text} />);
    fireEvent.click(getByText(text));
    expect(onClick).toHaveBeenCalled();
  });
});
