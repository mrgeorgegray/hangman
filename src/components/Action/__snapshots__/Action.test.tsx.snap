// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> renders 1`] = `
<div>
  <button
    class="css-554s8a-Action"
  >
    Click me
  </button>
</div>
`;
